onnx2kerastl/utils.py

```diff
@@ -131,3 +132,38 @@ def check_torch_keras_error(model, k_model, input_np, epsilon=1e-5, change_order
             max_error = error
 
     return max_error
+
+
+def match_tensors_rank(tensor_list, axis: int):
+    """
+    Adjusts the ranks of tensors of rank 1 in a given list to match the maximum rank by adding dummy dimensions
+    e.g., for input tensors shapes [(2,), (1, 4)] the unsqueezed tensors are [(1, 2), (1, 4)]
+
+    Args:
+        tensor_list (list): A list of tensors.
+
+    Returns:
+        list: A new list of tensors with adjusted ranks to match the maximum rank.
+              If all tensors in the input list already have the same rank, the original list is returned.
+    """
+    ranks = [tensor.shape.rank for tensor in tensor_list]
+    max_rank = max(ranks)
+
+    if len(set(ranks)) == 1:
+        return tensor_list
+    elif len(set(ranks)) > 2:
+        raise ValueError(f"More than 2 different ranks detected, broadcasting is ambiguous.\n"
+                         f"Check the outputs of layers: \n" + '\n'.join([tensor.name for tensor in tensor_list]))
```

Comment:
I think we can try to broadcast here multiple smaller ranks to the highest one no?

Reply:
I don't think so. If we have more than two different ranks than it becomes ambiguous which is the concatenation axis for each tensor, e.g., for tensor shapes (1, ), (1, 12), (2, 1, 4) and the concatenation axis is 1.
in this case how can we choose the reference tensor for the axis? should we broadcast the second tensor to (1, 12, 1) or to (1, 1, 12)?